### ISSUE TITLE
Adding retryStrategy to Template in DAG Breaks UI Layout

### ISSUE THREAD

**Issue by dsmithbauer** [argoproj/argo-workflows#880]
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

**What happened**:
When you add a retryStrategy to a container template referenced by a DAG, it causes the UI to flatten out such that it grows horizontally and the nodes of the graph are not connected to their dependent parents.

**What you expected to happen**:
DAG graph should maintain clean fan out, fan in structure with child nodes attached to dependent parents.

**How to reproduce it (as minimally and precisely as possible)**:

1. Run the examples/dag-diamond-steps.yaml workflow and note the clean and correct layout.
2. Add the following to the echo container template:
```
retryStrategy:
  limit: 1
```
3. Rerun the workflow and note how the graph starts each new level at the top and no longer maintains the correct, vertical layout.

**Anything else we need to know?**:

**Environment**:
- Argo version:
```
argo: v2.1.1
  BuildDate: 2018-05-29T20:39:36Z
  GitCommit: ac241c95c13f08e868cd6f5ee32c9ce273e239ff
  GitTreeState: clean
  GitTag: v2.1.1
  GoVersion: go1.9.3
  Compiler: gc
  Platform: linux/amd64
```
- Kubernetes version :
```
clientVersion:
  buildDate: 2017-10-24T19:48:57Z
  compiler: gc
  gitCommit: bdaeafa71f6c7c04636251031f93464384d54963
  gitTreeState: clean
  gitVersion: v1.8.2
  goVersion: go1.8.3
  major: "1"
  minor: "8"
  platform: linux/amd64
serverVersion:
  buildDate: 2018-05-21T09:05:37Z
  compiler: gc
  gitCommit: 2bba0127d85d5a46ab4b778548be28623b32d0b0
  gitTreeState: clean
  gitVersion: v1.10.3
  goVersion: go1.9.3
  major: "1"
  minor: "10"
  platform: linux/amd64
```

**Other debugging information (if applicable)**:
N/A

**Comment by ankushagarwal** [comment]
Happens to me too.

**Comment by jessesuen** [comment]
Thanks for the test case. Working on it.

**Comment by jessesuen** [comment]
Fixed. Fix will be available in v2.2.0
